### PR TITLE
feat(stats): add BME time series graph data endpoints

### PR DESCRIPTION
**Pull request by baktun14**
## Why

Refs CON-35

Add graph data API endpoints for BME (Burn & Mint Economics) metrics so the stats frontend can render time series charts for AKT burn, ACT mint, remint, vault balance, and collateral ratio.

## What

- Add new graph data names: `totalAktBurnedForAct`, `dailyAktBurnedForAct`, `totalActMinted`, `dailyActMinted`, `totalActBurnedForAkt`, `dailyActBurnedForAkt`, `totalAktReminted`, `dailyAktReminted`, `netAktBurned`, `dailyNetAktBurned`, `outstandingAct`, `vaultAkt`, `collateralRatio`, `gpuUtilization`
- Refactor `getBlockGraphData` from ~110-line method into smaller focused functions:
  - Unified `blockMetrics` config map replaces switch statement + separate `bmeMetrics`
  - `fetchDailyBlockSnapshots` — DB query + mapping
  - `buildGraphDataResponse` — response construction with dashboard lookup
  - `stripLeadingZeros` / `toRelativeValues` — pure standalone helpers
- Extract `AuthorizedGraphDataName` types into dedicated `stats.types.ts`
- Delete legacy `statsService.ts` in favor of the refactored `StatsService`
- Add functional tests and seeders for BME graph data

## Summary by CodeRabbit

* **New Features**
  * Added BME status history endpoint returning chronological status changes with collateral ratio data.

* **Improvements**
  * More accurate dashboard metrics, provider earnings, and GPU/utilization reporting via centralized data queries.
  * Provider dashboard metrics and lease/earnings calculations updated for reliability.

* **Refactor**
  * Reorganized stats into a repository-backed service and introduced typed graph-data identifiers.

* **Tests**
  * Expanded and modernized tests covering graph data, metrics routing, and edge cases.

### CODE REVIEW

File: apps/api/src/provider/services/provider-dashboard/provider-dashboard.service.ts [hunk 1]
```diff
@@ -2,15 +2,24 @@ import { Block } from "@akashnetwork/database/dbSchemas";
...
-import { getProviderEarningsAtHeight } from "@src/services/db/statsService";
-import { getProviderActiveResourcesAtHeight, getProviderTotalLeaseCountAtHeight } from "@src/services/db/statsService";
+import { ProviderEarningsService } from "@src/provider/services/provider-earnings/provider-earnings.service";
 
 @singleton()
 export class ProviderDashboardService {
+  readonly #chainDb: Sequelize;
+
+  constructor(
+    @inject(CHAIN_DB) chainDb: Sequelize,
```
Comment (stalniy) [inline apps/api/src/provider/services/provider-dashboard/provider-dashboard.service.ts:17]: **thought:** this looks like repository level, not a service level thing.

File: apps/api/src/provider/services/provider-earnings/provider-earnings.service.ts [hunk 2]
```diff
@@ -48,4 +54,58 @@ export class ProviderEarningsService {
...
+    WHERE d."lastBlockHeightYet" >= :providerCreatedHeight AND d."firstBlockHeight" <= :height
+    ORDER BY d.date DESC
+`,
+      {
+        type: QueryTypes.SELECT,
+        replacements: { provider, height, providerCreatedHeight }
+      }
+    );
+
+    return days.reduce(
```
Comment (stalniy) [inline apps/api/src/provider/services/provider-earnings/provider-earnings.service.ts:101]: **question:** how big `days` array can be? All loops is a potential issue for event loop delay. So, need to take this into account

**Review by stalniy — APPROVED** [review]

File: apps/api/src/provider/services/provider-dashboard/provider-dashboard.service.ts (hunk 1 above)
Comment (baktun14) [reply apps/api/src/provider/services/provider-dashboard/provider-dashboard.service.ts:17]: Yea I moved it there from another file from legacy, I can refactor it

File: apps/api/src/provider/services/provider-earnings/provider-earnings.service.ts (hunk 2 above)
Comment (baktun14) [reply apps/api/src/provider/services/provider-earnings/provider-earnings.service.ts:101]: The `days` array is bounded by the number of calendar days between the provider's creation height and the target height. For a provider that's been active for 1 year, that's ~365 entries — shouldn't cause event loop issues.

Also added nullish coalescing (`?? 0`) for `totalUAkt`/`totalUUsdc` to guard against NaN from LEFT JOIN nulls. Fixed in 94f099c.

File: apps/api/src/dashboard/repositories/stats/stats.repository.ts [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+            ps."activeGPU", ps."pendingGPU", ps."availableGPU",
+            ps."isOnline",
+            n.id as "nodeId"
+          FROM "providerSnapshot" ps
+          INNER JOIN "provider" ON "provider"."owner"=ps."owner"
+          INNER JOIN "providerSnapshotNode" n ON n."snapshotId"=ps.id AND n."gpuAllocatable" > 0
+          LEFT JOIN "providerSnapshotNodeGPU" gpu ON gpu."snapshotNodeId" = n.id
+          WHERE ps."isLastSuccessOfDay" = TRUE
+          ORDER BY "hostUri",DATE("checkDate"),"checkDate" DESC
+        ) "dailyProviderStats"
```
[the remark this answers is not in the document]
Comment (baktun14) [reply apps/api/src/dashboard/repositories/stats/stats.repository.ts:111]: This is a pre-existing bug — the query was moved as-is during the refactoring extraction. Filed as [CON-104](https://linear.app/akash-network/issue/CON-104) for a follow-up fix.

File: apps/api/src/provider/services/provider-earnings/provider-earnings.service.ts [hunk 4]
```diff
@@ -48,4 +54,58 @@ export class ProviderEarningsService {
...
+      (acc, d) => {
+        acc.uakt += d.totalUAkt ?? 0;
+        acc.uusdc += d.totalUUsdc ?? 0;
+        acc.uusd += (d.totalUAkt ?? 0) * d.aktPrice + (d.totalUUsdc ?? 0);
+        return acc;
+      },
+      { uakt: 0, uusdc: 0, uusd: 0 }
+    );
```
[the remark this answers is not in the document]
Comment (baktun14) [reply apps/api/src/provider/services/provider-earnings/provider-earnings.service.ts:109]: Fixed in 3d0f618c6 — added `?? 0` null guard on `aktPrice`.
